language-support/ts/daml-ledger/index.ts

```diff
@@ -587,13 +589,31 @@ class Ledger {
    * @typeparam T The contract template type.
    * @typeparam K The contract key type.
    * @typeparam I The contract id type.
+   *
    */
   streamQuery<T extends object, K, I extends string>(
     template: Template<T, K, I>,
     query?: Query<T>,
   ): Stream<T, K, I, readonly CreateEvent<T, K, I>[]> {
-    const request = {templateIds: [template.templateId], query};
-    const reconnectRequest = (): object[] => [request];
+    if (query === undefined) {
+      return this.streamQueryCommon(template, [], "streamQuery");
+    } else {
+      return this.streamQueryCommon(template, [query], "streamQuery");
+    }
+  }
+
+  /**
+   * @internal
+   *
+   */
+  streamQueryCommon<T extends object, K, I extends string>(
+    template: Template<T, K, I>,
+    queries: Query<T>[],
+    name: string,
+  ): Stream<T, K, I, readonly CreateEvent<T, K, I>[]> {
+    const request = queries.length == 0 ? [{templateIds: [template.templateId]}]
```

Comment:
nit: I prefer indentation that does not depend on identifier length so aline break after `?`.

Reply:
Fair point, I'll update that.